modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -80,9 +80,12 @@ interface DomParser {
   addNodeFilter: (name: string, callback: (nodes: AstNode[], name: string, args: ParserArgs) => void) => void;
   getNodeFilters: () => ParserFilter[];
   filterNode: (node: AstNode) => AstNode;
-  parse: (html: string, args?: ParserArgs) => AstNode;
+  // TODO: TINY-4627 (TINY-8190) make this back into an AstNode, but in a good way
+  parse: (html: string, args?: ParserArgs) => Node;
 }
 
+let lazyParser: DOMParser | null = null;
```

Comment:
Can you please change this to a `Thunk.cached` call instead? That's our normal way of doing something that has to lazily initialise and only run once.